### PR TITLE
[DEVOPS-1160] iohk-nix implementation

### PR DESCRIPTION
**Pull request by disassembler**
This PR CHANGES.

* Uses iohk-nix for shared nix library functions
* Converts installers to overlays
* Adds Phase Metrics to builds
* Potentially fixes nix-shell on OSX mojave (needs tested)
## Todo:

- [ ] Todo
- [ ] Todo
- [ ] Todo

## Screenshots:

- Screenshot

---

### Basics

- [ ] PR is updated to the most recent version of target branch (and there are no conflicts)
- [ ] PR has good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance tests are passing (`yarn run test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn run dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn run package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn run flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn run lint`)
- [ ] Text changes are proofread and approved (Jane Wild)
- [ ] There are no missing translations (running `yarn run manage:translations` produces no changes)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn run storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly documented and commented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-rendering
- [ ] Any code that only works in Electron is neatly separated from components

### Testing
- [ ] New feature / change is covered by acceptance tests
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature / change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review:
- [ ] Merge PR
- [ ] Delete source branch
- [ ] Move ticket to `done` on the Youtrack board

### CODE REVIEW

**Comment by disassembler** [comment]
Work in progress still. Need to still get the release.nix correct so everything gets built and cached by hydra.

**Comment by disassembler** [comment]
bors ping

**Comment by disassembler** [comment]
bors try

**Comment by disassembler** [comment]
bors try

File: shell.nix [hunk 1]
```diff
@@ -67,6 +67,18 @@ let
     rm $out/wallet-topology.yaml
     cp ${demoTopologyYaml} $out/wallet-topology.yaml
   '';
+  buildShell = pkgs.stdenv.mkDerivation {
```
Comment (rvl) [inline shell.nix]: Maybe add a comment about what this is -- I suppose it's everything required for the `build-installer-unix.sh` script. Does it need the `autoStartBackend` conditional?

File: installers/default.nix [hunk 2]
```diff
@@ -1,66 +1,54 @@
...
-      github = self.callPackage ./github.nix {};
-    };
+  inherit daedalus-bridge;
+  addTestStubsOverlay = import ./overlays/add-test-stubs.nix {
+    inherit pkgs daedalus-bridge;
+  };
+  haskellPackages = callPackage localLib.iohkNix.haskellPackages {
+    inherit forceDontCheck enableProfiling enablePhaseMetrics enableHaddockHydra
+      enableBenchmarks fasterBuild enableDebugging enableSplitCheck;
+    pkgsGenerated = haskell.packages.ghc822;
```
Comment (angerman) [inline installers/default.nix]: I know this is basically just a rewrite from the previous. But why 822, aren't we mostly on 843 now?

**Pull request merged**